CMakeLists.txt

```diff
@@ -399,6 +404,11 @@ if(ANDROID)
 	SET(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
 	SET(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
 	SET(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
+elseif(HAIKU)
+	set(USE_X11 0)
+	set(USE_UPNP 0)
+	set(USE_EGL 0)
+	list(APPEND LIBS network)
```

Comment:
Then add the dependency to "network" directly in the core if that's where it is needed, don't add it globally.
That LIBS variable needs to be removed and explicit direct dependencies on each target should be added instead.